src/libbson/tests/test-bson.c

```diff
@@ -2461,6 +2462,46 @@ test_bson_append_null_from_utf8_or_symbol (void) {
    bson_destroy (&bson);
 }
 
+static void
+test_bson_as_json_string (void)
+{
+   bson_t *all_types;
+   int i;
+   char *actual;
+   const char *expected =
+      "{ \"double\" : { \"$numberDouble\" : \"1.0\" }, \"string\" : "
+      "\"string_example\", \"document\" : { \"x\" : \"y\" }, \"document\" : [ "
+      "\"x\" ], \"binary\" : { \"$binary\" : { \"base64\": \"ZGF0YQ==\", "
+      "\"subType\" : \"00\" } }, \"undefined\" : { \"$undefined\" : true }, "
+      "\"oid\" : { \"$oid\" : \"000000000000000000000000\" }, \"bool\" : true, "
+      "\"datetime\" : { \"$date\" : { \"$numberLong\" : \"123\" } }, \"null\" "
+      ": null, \"regex\" : { \"$regularExpression\" : { \"pattern\" : \"a+\", "
+      "\"options\" : \"\" } }, \"dbpointer\" : { \"$dbPointer\" : { \"$ref\" : "
+      "\"collection\", \"$id\" : { \"$oid\" : \"000000000000000000000000\" } } "
+      "}, \"code\" : { \"$code\" : \"var x = 1;\" }, \"symbol\" : { "
+      "\"$symbol\" : \"symbol_example\" }, \"code\" : { \"$code\" : \"var x = "
+      "1;\" }, \"code_w_scope\" : { \"$code\" : \"var x = 1;\", \"$scope\" : { "
+      "} }, \"int32\" : { \"$numberInt\" : \"1\" }, \"timestamp\" : { "
+      "\"$timestamp\" : { \"t\" : 2, \"i\" : 3 } }, \"int64\" : { "
+      "\"$numberLong\" : \"4\" }, \"decimal128\" : { \"$numberDecimal\" : "
```

Comment:
This tests the bug at hand, that an int64 produces `{ "$numberLong": "4" }` with the space before `}`. But it seemed prudent to test JSON produced of all BSON types while adding this.